doc/API.md

```diff
@@ -47,21 +48,160 @@ async ({ id, object }) => {
 });
 ```
 
-## Function with guards
+## Extended function declaration
 
-Use guards optionally. You can apply guard for one or a few arguments. You can
-apply it multiple times for one argument for complex checking or for combination
-of multiple arguments.
+Declaration elements (fields):
+- `caption` - method display name
+- `description` - method description
+- `parameters` (optional) - parameters declarative schema
+- `validate` (optional) - parameters imperative validation function
+- `timeout` (optional) - execution timeout in milliseconds
+- `method` - sync or async lambda implementing method
+- `returns` (optional) - returning value declarative schema
+- `assert` (optional) - returning value imperative assertion function
+- `examples` - array of examples
+- `example` - single example (shorthand)
+- `errors` - collection of possible error
 
 ```js
-async ({ name, city, age }) => {
-  name = guard(name, { type: 'string', default: 'Marcus' });
-  city = guard(city, { type: 'string', default: 'Roma' });
-  age = guard(age, { type: 'number', required: true });
+({
+  parameters: {
+    a: { type: 'number' },
```

Comment:
Not sure if we already have this somewhere, but I think it would be better to have types connected with validation, i.e. each type will have a default validator which will always be executed.